modules/scip/9.2.0.bcr.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,161 @@
++# limitations under the License.
++
++exports_files(
++    ["src/lpi/lpi_glop.cpp"],
++)
++
++config_setting(
++    name = "on_linux",
++    constraint_values = [
++        "@platforms//os:linux",
++    ],
++)
++
++config_setting(
++    name = "on_macos",
++    constraint_values = [
++        "@platforms//os:macos",
++    ],
++)
++
++config_setting(
++    name = "on_windows",
++    constraint_values = [
++        "@platforms//os:windows",
++    ],
++)
++
++PLATFORM_FLAGS = select({
++    "on_linux": [
++        "-Wunknown-pragmas",
++        "-fexceptions",
++        "-DSYM=bliss",
++    ],
++    "on_macos": [
++        "-Wunknown-pragmas",
++        "-fexceptions",
++        "-DSYM=bliss",
++    ],
++    "on_windows": [
++        "/DSYM=none",
++        "/DSCIP_NO_SIGACTION",
++        "/DSCIP_NO_STRTOK_R",
++    ],
++    "//conditions:default": [],
++})
++
++PLATFORM_DEPS = select({
++    "on_linux": ["@bliss"],
++    "on_macos": ["@bliss"],
++    "on_windows": [],
++    "//conditions:default": [],
++})
++
++BLISS_FILE = select({
++    "on_linux": ["src/symmetry/compute_symmetry_bliss.cpp"],
++    "on_macos": ["src/symmetry/compute_symmetry_bliss.cpp"],
++    "on_windows": ["src/symmetry/compute_symmetry_none.cpp"],
++    "//conditions:default": ["src/symmetry/compute_symmetry_none.cpp"],
++})
++
++BASE_SRCS = glob(
++    [
++        "src/*/*.c",
++    ],
++    exclude = [
++        "src/lpi/lpi_*.c",
++        "src/nauty/*",
++        "src/scip/exprinterpret_*.c",
++        "src/scip/nlpi_filtersqp.c",
++        "src/scip/nlpi_worhp.c",
++        "src/scip/*_xyz.c",
++        "src/scip/sorttpl.c",
++        "src/symmetry/compute_symmetry_*.cpp",
++        "src/symmetry/*nauty*",
++        "src/tpi/tpi_*.c",
++    ],
++)
++
++BASE_HDRS = glob(
++    [
++        "src/*/*.h",
++        "src/*/*.hpp",
++        "src/scip/githash.c",
++        "src/scip/sorttpl.c",
++        "src/scip/buildflags.c",
++    ],
++    exclude = [ "src/scip/*_xyz.h" ],
++)
++
++DEFINES = [
++    # Scip v800 optionally depends on scip/config.h and
++    # scip/scip_export.h that are generated by build system.
++    #
++    # We need every library and binary that depends on SCIP libraries to
++    # define this macro. That is why we use `defines' here instead of
++    # `copts' or `local_defines'.
++    "NO_CONFIG_HEADER",
++]
++
++cc_library(
++    name = "scip",
++    srcs =  BASE_SRCS + BLISS_FILE + [
++        "src/scip/exprinterpret_none.c",
++        "src/tpi/tpi_none.c",
++    ],
++    hdrs = BASE_HDRS,
++    copts = [
++        "$(STACK_FRAME_UNLIMITED)",  # src/scip/reader_cnf.c
++        "-DTPI_NONE",  # src/tpi/type_tpi_tnycthrd.h
++        "-DNPARASCIP",
++    ] + PLATFORM_FLAGS,
++    defines = DEFINES,
++    features = ["-parse_headers"],
++    includes = [
++        "src",
++    ],
++    visibility = ["//visibility:public"],
++    deps = [
++        "@zlib",
++    ] + PLATFORM_DEPS,
++)
++
++cc_library(
++    name = "parallel_scip",
```

Comment:
Nit: Maybe `scip_parallel` to find this as well with a prefix search.

Reply:
Done